client/landing/stepper/declarative-flow/internals/steps-repository/trial-acknowledge/index.tsx

```diff
@@ -1,11 +1,31 @@
 import { StepContainer, isNewHostedSiteCreationFlow } from '@automattic/onboarding';
+import { useEffect } from 'react';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useDispatch, useSelector } from 'calypso/state';
+import { getCurrentUserId, isCurrentUserEmailVerified } from 'calypso/state/current-user/selectors';
+import { savePreference } from 'calypso/state/preferences/actions';
 import { HostingTrialAcknowledge } from './hosting-trial-acknowledge';
 import { MigrationTrialAcknowledge } from './migration-trial-acknowledge';
 import type { Step } from 'calypso/landing/stepper/declarative-flow/internals/types';
 import './style.scss';
+
 const TrialAcknowledge: Step = function TrialAcknowledge( { navigation, flow, stepName } ) {
 	const { goBack } = navigation;
+	const dispatch = useDispatch();
+	const userId = useSelector( getCurrentUserId );
+	const isEmailVerified = useSelector( isCurrentUserEmailVerified );
+	const pathStep = isEmailVerified ? null : `/setup/${ flow }/${ stepName }`;
+
+	/**
+	 * Save the selected trial plan path in the user's preferences.
+	 */
+	const saveHostingFlowPathStep = () => {
+		dispatch( savePreference( `hosting-flow-path-step-${ userId }`, pathStep ) );
+	};
+
+	useEffect( () => {
+		saveHostingFlowPathStep();
+	}, [ pathStep ] );
```

Comment:
is there a reason we are not checking if this is a hosting flow here? (as we did in the other 2 files)

Reply:
From what I can tell, this step is only relevant on hosting flow so seemed redundant.

Reply:
Should be fixed in new hook